#### Задача 1. 
Даны две независимые выборки. Не соблюдается условие нормальности
x1 380,420, 290
y1 140,360,200,900
Сделайте вывод по результатам, полученным с помощью функции, имеются ли статистические различия между группами?

In [40]:
import numpy as np
from scipy import stats

In [41]:
x1 = np.array([380,420, 290])
y1 = np.array([140,360,200,900])
stats.mannwhitneyu(x1, y1)

MannwhitneyuResult(statistic=8.0, pvalue=0.6285714285714286)

pvalue=0.6285714285714286 > alfa=0.05. Значит Н0 не отвергается при уровне значимости 0,05. Статистических различий нет

#### Задача 2. 
Исследовалось влияние препарата на уровень давления пациентов. Сначала
измерялось давление до приема препарата, потом через 10 минут и через 30 минут. Есть
ли статистически значимые различия между измерениями давления? В выборках не соблюдается условие нормальности.
1е измерение до приема препарата: 150, 160, 165, 145, 155
2е измерение через 10 минут: 140, 155, 150, 130, 135
3е измерение через 30 минут: 130, 130, 120, 130, 125

In [42]:
x1 = np.array([150, 160, 165, 145, 155])
x2 = np.array([140, 155, 150, 130, 135])
x3 = np.array([130, 130, 120, 130, 125])
stats.friedmanchisquare(x1, x2, x3)


FriedmanchisquareResult(statistic=9.578947368421062, pvalue=0.00831683351100441)

pvalue=0.00831683351100441 < alfa = 0.05. Значит Н0 отвергаем при уровне значимости 0,05. Препарат оказывает влияние на давление пациентов

#### Задача 3. 
Сравните 1 и 2 е измерения, предполагая, что 3го измерения через 30 минут не было. Есть
ли статистически значимые различия между измерениями давления?

In [43]:
x1 = np.array([150, 160, 165, 145, 155])
x2 = np.array([140, 155, 150, 130, 135])
stats.wilcoxon(x1, x2)

WilcoxonResult(statistic=0.0, pvalue=0.0625)

pvalue=0.0625 > alfa = 0.05. Значит Н0 не отвергаем при уровне значимости 0,05. Препарат не оказывает влияние на давление пациентов

#### Задача 4. 
Даны 3 группы учеников плавания. Не соблюдается условие нормальности.
В 1 группе время на дистанцию 50 м составляют:
56, 60, 62, 55, 71, 67, 59, 58, 64, 67
Вторая группа : 57, 58, 69, 48, 72, 70, 68, 71, 50, 53
Третья группа: 57, 67, 49, 48, 47, 55, 66, 51, 54
Есть
ли статистически значимые различия между группами?

In [44]:
x1 = np.array([56, 60, 62, 55, 71, 67, 59, 58, 64, 67])
x2 = np.array([57, 58, 69, 48, 72, 70, 68, 71, 50, 53])
x3 = np.array([57, 67, 49, 48, 47, 55, 66, 51, 54])
stats.kruskal(x1, x2, x3)


KruskalResult(statistic=5.465564058257224, pvalue=0.06503809985904942)

pvalue=0.06503809985904942 > alfa = 0.05. Значит Н0 не отвергаем при уровне значимости 0,05. Статистически значимых различий между группами нет

#### Задача 5. 
Заявляется, что партия изготавливается со средним арифметическим 2,5 см. Проверить
данную гипотезу, если известно, что размеры изделий подчинены нормальному закону
распределения. Объем выборки 10, уровень статистической значимости 5%
2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34

In [45]:
alfa = 0.05
M = 2.5
x = np.array([2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34])
n = len(x)
x_mean = np.mean(x)
x_std = np.std(x, ddof = 1)
t = (x_mean - M) / (x_std / n ** 0.5)
t


0.5630613661802959

In [46]:
t1 = stats.t.ppf(alfa / 2, df = n - 1)
t2 = stats.t.ppf(1 - alfa / 2, df = n - 1)
t1, t2

(-2.262157162740992, 2.2621571627409915)

t попадает внутрь интервала между t1 и t2. Н0 не отвергается. Партия изготавливается со средним арифметическим 2,5 см